orderer/consensus/etcdraft/chain.go

```diff
@@ -911,6 +921,41 @@ func (c *Chain) ordered(msg *orderer.SubmitRequest) (batches [][]*common.Envelop
 			}
 		}
 
+		if !c.leaderTransferFailed {
+			if c.checkForEvictionNCertRotation(msg.Payload) {
+				go func() {
+					if transferee, err := c.Node.abdicateLeader(c.raftID); err != nil {
+						if err.Error() == LEADERTRANSFERTIMEOUT.Error() && transferee != raft.None {
+							if transferee != c.leadershipTransferCandidate {
```

Comment:
I don't understand why we need this transfer candidate and transfer failed fields? 

Can't we simply this?

```
		if c.checkForEvictionNCertRotation(msg.Payload) {
			if ! atomic.CompareAndSwapUint32(&c.leadershipTransferInProgress, 0, 1) {
				return nil, false, errors.Errorf("leadership transfer is currently in progress")
			}
			go func() {
				
				defer func() {
					atomic.StoreUint32(&c.leadershipTransferInProgress, 0)	
				}()
				
				if transferee, err := c.Node.abdicateLeader(c.raftID); err != nil {
					... 

``

Reply:
These 2 variables need to control the flow wrt the leadership timeout and retries. 
it would limit the leadership transfer attempt for a given node not more than 2 times.